map/mapObject.mjs

```diff
@@ -0,0 +1,102 @@
+"use strict"
+export {MapObject, MapIcon}
+
+import {worldSpaceToMapSpace, mapSpaceToScreenSpace, iconH} from "../map.js"
+import {Point} from "./point.mjs"
+
+/**
+ * An object that will be displayed on the map canvas.
+ */
+function MapObject(){
+    this.minX = 0;
+    this.maxX = 0;
+    this.minY = 0;
+    this.maxY = 0;
+}
+/**
+ * Does this object contain the specified point?
+ * @param {Point} point point to check
+ */
+MapObject.prototype.contains = function(point){
+    if(point == null){
+        return false;
+    }
+    return (this.minX < point.x && point.x < this.maxX && this.minY < point.y && point.y < this.maxY);
+}
+MapObject.prototype.width = function(){
+    return this.maxX - this.minX;
+}
+MapObject.prototype.height = function(){
+    return this.maxY - this.minY;
+}
```

Comment:
So this is basically how you add methods to a "class" in javascript. put them on the prototype of the constructor.

Reply:
These methods are the main reason why classes are being used. the contains() function is very useful.